src/codelabs/iterables.md

```diff
@@ -0,0 +1,615 @@
+also be read as `Iterable`, with the properties `entries` or `values`.
+
+## What is an Iterable?
+
+An `Iterable` is a collection of elements that can be accessed sequentially.
+
+`Iterable`, in Dart, is an abstract class, meaning that you can't instantiate it directly.
+However you can create a new `Iterable` by creating a new `List` or a `Set`.
+ 
+This example shows a `List` of `int`, which is also an `Iterable` of `int`:
+
+{% prettify dart %}
+    Iterable<int> iterable = [1, 2, 3];
+{% endprettify %}
+
+The difference with a `List`, is that with the `Iterable` you can't 
+guarantee that reading elements by index will be efficient.
+`Iterable`, as opposed as `List`, does not have the `[]` operator.
+
+For example **the following code will give you an error**:
+
+{:.bad}
+{% prettify dart %}
+    Iterable<int> iterable = [1, 2, 3];
+    var value = iterable[![1]!];
+{% endprettify %}
+
+If you read elements with `[]`, the compiler will tell you that
+the operator `'[]'` isn't defined for the class `Iterable` which
+means that you can't use `[index]` in this case.
+
+You can instead read elements with `elementAt()`, which will step through
+the elements of the iterable until reaching that position.
+
+{% prettify dart %}
+    Iterable<int> iterable = [1, 2, 3];
+    var value = iterable.elementAt(1);
+{% endprettify %}
+
+Continue to the next section to learn more about how to access elements of an `Iterable`.
+
+## Reading elements
+
+The elements of an `Iterable` can be read sequentially.
+
+One way to do that, is by getting an `Iterator` using the `iterator` getter,
+and using it to step through the values. However, it is easier to use the 
+`for-in loop`, which internally uses an `Iterator` too.
+
+### Example: Using for-in loop
+
+The following example shows how to read elements using  a `for-in loop`.
+
+[//]: https://gist.github.com/597f152c1df414fc4d7d78d1be22ac22
+<iframe
+  src="{{site.dartpad-embed}}?id=597f152c1df414fc4d7d78d1be22ac22"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="200"
+  width="100%" >
+</iframe>
+
+The `for-in loop`, as opposed to the `Iterator`, makes your code easier
+to read and understand, and it is less prone to errors.
+
+Here's an example with `Iterator`:
+
+{% prettify dart %}
+var it = iterable.iterator;	
+while (it.moveNext()) {	
+  print(it.current);	
+}
+{% endprettify %}
+
+{{site.alert.info}}
+Stepping with the `Iterator` is done
+by calling `Iterator.moveNext`, and if the call returns true, the `Iterator` 
+has now moved to the next element, which is then available as `Iterator.current`. 
+
+If the call returns `false`, there are no more elements, and `iterator.current` returns `null`.
+{{site.alert.end}}
+
+{{site.alert.secondary}}
+**Key terms:**
+* **Iterable**: the Dart [Iterable][iterable class] class.
+* **Iterator**: used to read elements from an Iterable.
```

Comment:
->

* **Iterator**: An object used by `for-in` to read elements from an `Iterable` object.